clientapi/producers/syncapi.go

```diff
@@ -173,3 +175,21 @@ func (p *SyncAPIProducer) SendTyping(
 	_, err := p.JetStream.PublishMsg(m, nats.Context(ctx))
 	return err
 }
+
+func (p *SyncAPIProducer) SendPresence(
+	ctx context.Context, userID, presence string, statusMsg *string,
+) error {
+	m := nats.NewMsg(p.TopicPresenceEvent)
+	m.Header.Set(jetstream.UserID, userID)
+	m.Header.Set("presence", presence)
+	nilMsg := statusMsg == nil
+	if !nilMsg {
+		m.Header.Set("status_msg", *statusMsg)
+	}
+	m.Header.Set("status_msg_nil", strconv.FormatBool(nilMsg))
```

Comment:
Do we need to do this? I would naively expect that a lack of the `status_msg` header on the receiving side could be considered as a `nil` status message.

Reply:
Hm.. Having tested a bit and switching everything to something like
```go
var statusMsg *string = nil
if data, ok := m.Header["status_msg"]; ok && len(data) > 0 {
  statusMsg = &data[0]
}
```
(`m.Header.Get("status_msg")` returns empty strings..) and
```go
if  statusMsg != nil {
   m.Header.Set("status_msg", *statusMsg)
}
```
It doesn't seem to work as expected.